tests/airmar_tests/test_nmeaparser.py

```diff
@@ -41,3 +41,143 @@ def test_settings(self):
     def test_checksum(self):
         """ Tests checksum function """
         self.assertEqual(self.parser.checksum("test"), "16")
+
+    def _make_nmea_sentence(self, sentence):
+        """ Helper function to create custom nmea sentences 
+        
+        Keyword Arguments:
+        sentence -- The nmea sentence body to create nmea sentence from.
+
+        Returns:
+        A properly formatted NMEA sentence with checksum.
+        """
+        return "$" + sentence + "*{}\r\n".format(self.parser.checksum(sentence))
+
+    def test_update_sentence_gpdtm(self):
+        """ Tests format fields """
+        data = {} # initialize raw data
+
+        # Test GPDTM sentences
+        sentence = "GPDTM,W84,,100.001,N,200.001,E,10,W84"
```

Comment:
it may be simpler to parse by saving each string field in a list, making your sentence using string formatting, and then using the list to assertEqual

Reply:
Good idea, forgot I could do it this way. Completed.